crates/ruff/src/rules/flake8_type_checking/rules/runtime_import_in_type_checking_block.rs

```diff
@@ -75,12 +89,45 @@ pub(crate) fn runtime_import_in_type_checking_block(
                 .is_runtime()
         })
     {
-        let diagnostic = Diagnostic::new(
+        let mut diagnostic = Diagnostic::new(
             RuntimeImportInTypeCheckingBlock {
                 full_name: full_name.to_string(),
             },
             binding.range,
         );
+
+        if checker.patch(diagnostic.kind.rule()) {
+            diagnostic.try_set_fix(|| {
+                // Step 1) Remove the import from the type-checking block.
+                let source = binding.source.unwrap();
+                let deleted: Vec<&Stmt> = checker.deletions.iter().map(Into::into).collect();
+                let stmt = checker.semantic_model().stmts[source];
+                let parent = checker
+                    .semantic_model()
+                    .stmts
+                    .parent_id(source)
+                    .map(|id| checker.semantic_model().stmts[id]);
+                let remove_import_edit = autofix::edits::remove_unused_imports(
+                    std::iter::once(full_name),
+                    stmt,
+                    parent,
+                    &deleted,
+                    checker.locator,
+                    checker.indexer,
+                    checker.stylist,
+                )?;
+
+                // Step 2) Add the import to the top-level.
+                let reference = checker.semantic_model().references.resolve(*reference_id);
+                let add_import_edit = checker.importer.to_runtime_import(
+                    &StmtImport { stmt, full_name },
+                    reference.range().start(),
+                )?;
+
+                Ok(Fix::suggested_edits(remove_import_edit, [add_import_edit]).sorted())
```

Comment:
I think I would prefer to always store the `Edit`s in sorted order if that's what we need rather than requiring fix authors to remember calling `sorted`.